src/Nethermind/Nethermind.Merge.Plugin/Handlers/GetPayloadBodiesByRangeV2Handler.cs

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Nethermind.Blockchain;
+using Nethermind.Core;
+using Nethermind.Core.ConsensusRequests;
+using Nethermind.JsonRpc;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.Data;
+using Nethermind.Merge.Plugin.Handlers;
+
+namespace Nethermind.Merge.Plugin.handlers;
+
+public class GetPayloadBodiesByRangeV2Handler(IBlockTree blockTree, ILogManager logManager) : GetPayloadBodiesByRangeV1Handler(blockTree, logManager), IGetPayloadBodiesByRangeV2Handler
+{
+    public new Task<ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>> Handle(long start, long count)
+    {
+        if (start < 1 || count < 1)
+        {
+            var error = $"'{nameof(start)}' and '{nameof(count)}' must be positive numbers";
+
+            if (_logger.IsError) _logger.Error($"{nameof(GetPayloadBodiesByRangeV2Handler)}: ${error}");
+
+            return ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>.Fail(error, ErrorCodes.InvalidParams);
+        }
+        if (count > MaxCount)
+        {
+            var error = $"The number of requested bodies must not exceed {MaxCount}";
+
+            if (_logger.IsError) _logger.Error($"{nameof(GetPayloadBodiesByRangeV2Handler)}: {error}");
+
+            return ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>.Fail(error, MergeErrorCodes.TooLargeRequest);
+        }
+
+        return Task.FromResult(ResultWrapper<IEnumerable<ExecutionPayloadBodyV2Result?>>.Success(GetRequests(start, count)));
+    }
+
+
+    private IEnumerable<ExecutionPayloadBodyV2Result?> GetRequests(long start, long count)
+    {
+        var headNumber = _blockTree.Head?.Number ?? 0;
+        for (long i = start, c = Math.Min(start + count - 1, headNumber); i <= c; i++)
+        {
+            Block? block = _blockTree.FindBlock(i);
+
+            ConsensusRequest[]? consensusRequests = block?.Requests;
+
+            List<Deposit> deposits = [];
```

Comment:
You can do it without lists, chec ExecutionPayloadV4, pls think if the code could be shared